[Release 1.5] Upgrade notice for new authentication APIs (#6615)
istio/istio.io: content/en/news/releases/1.5.x/announcing-1.5/upgrade-notes/index.md

preserved but new behavior was introduced that would be surprising to someone
familiar with the use and operation of Istio 1.4.

## Authentication policy

> Istio 1.5 introduces [`PeerAuthentication`](/docs/reference/config/security/todo-link/) and [`RequestAuthentication`](/docs/reference/config/security/todo-linke/) that is going to replace alpha Authentication API. For more information how to use the new API, see the [authentication policy](/docs/tasks/security/authentication/authn-policy) tutorial.

Reviewer: ```suggestion
Istio 1.5 introduces [`PeerAuthentication`](/docs/reference/config/security/todo-link/) and [`RequestAuthentication`](/docs/reference/config/security/todo-link/), which are replacing the alpha version of the Authentication API. For more information about how to use the new API, see the [authentication policy](/docs/tasks/security/authentication/authn-policy) tutorial.
```

Reviewer: Also, can you clarify the timing? Does the new API replace the old API effective immediately? Is the old API now deprecated?

* After upgrading Istio, existing alpha authentication policies still in use.
* You should start replacing alpha policies with the equivalent `PeerAuthentication` and `RequestAuthentication`. It is recommended to start with the most specific scope (workload-wide), then namespace-wide and finally mesh-wide.
* Once the new API is added for workload/namespace/mesh, alpha API will be ignored in that scope and can be safely remove.

> * After finish migrating to new authentication APIs, you can delete all alpha policies, if you haven't done so, with this command:

Reviewer: ```suggestion

* After you upgrade Istio, your alpha authentication policies remain in place until you replace them with the equivalent `PeerAuthentication` and `RequestAuthentication`. We recommend starting with workload-wide (the most specific scope), then namespace-wide, and finally mesh-wide.
* After you replace policies for workload, namespace, and mesh, you can safely remove the alpha authentication policies. To delete the alpha policies, use this command:
```
